<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LLMConfigWidget</class>
 <widget class="QWidget" name="LLMConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>400</height>
   </rect>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="triggerWordLabel">
     <property name="text">
      <string>Trigger Word:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="triggerWordEdit">
     <property name="toolTip">
      <string>The word to trigger the LLM runner (e.g., 'llm')</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="providerLabel">
     <property name="text">
      <string>Provider:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QComboBox" name="providerCombo">
     <property name="toolTip">
      <string>Select your LLM provider</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="apiKeyLabel">
     <property name="text">
      <string>API Key:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QLineEdit" name="apiKeyEdit">
     <property name="echoMode">
      <enum>QLineEdit::Password</enum>
     </property>
     <property name="toolTip">
      <string>Your API key from the provider</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="baseUrlLabel">
     <property name="text">
      <string>Base URL:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QLineEdit" name="baseUrlEdit">
     <property name="placeholderText">
      <string>http://localhost:11434/v1/chat/completions</string>
     </property>
     <property name="toolTip">
      <string>API endpoint URL (only for Local LLM)</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="modelLabel">
     <property name="text">
      <string>Model:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLineEdit" name="modelEdit">
     <property name="toolTip">
      <string>The model to use (e.g., gpt-4, claude-3-5-sonnet-20241022)</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="maxTokensLabel">
     <property name="text">
      <string>Max Tokens:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QSpinBox" name="maxTokensSpin">
     <property name="minimum">
      <number>50</number>
     </property>
     <property name="maximum">
      <number>4000</number>
     </property>
     <property name="value">
      <number>150</number>
     </property>
     <property name="toolTip">
      <string>Maximum number of tokens in the response</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="timeoutLabel">
     <property name="text">
      <string>Timeout (seconds):</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QSpinBox" name="timeoutSpin">
     <property name="minimum">
      <number>5</number>
     </property>
     <property name="maximum">
      <number>120</number>
     </property>
     <property name="value">
      <number>30</number>
     </property>
     <property name="toolTip">
      <string>Request timeout in seconds</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="debounceDelayLabel">
     <property name="text">
      <string>Debounce Delay (ms):</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QSpinBox" name="debounceDelaySpin">
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>5000</number>
     </property>
     <property name="value">
      <number>800</number>
     </property>
     <property name="toolTip">
      <string>Delay before sending query to LLM after user stops typing (in milliseconds)</string>
     </property>
    </widget>
   </item>
   <item row="8" column="0" colspan="2">
    <widget class="QLabel" name="infoLabel">
     <property name="text">
      <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;&lt;b&gt;Usage:&lt;/b&gt; Type your trigger word followed by your question in KRunner.&lt;/p&gt;&lt;p&gt;Example: &lt;i&gt;llm what is the capital of France?&lt;/i&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
